packages/toolpad-app/src/runtime/evalJsBindings.ts

```diff
@@ -249,12 +249,13 @@ export default function evalJsBindings(
             Array.isArray(value) ? `[${nestedPropName}]` : `.${nestedPropName}`
           }`;
 
-          const nestedBindingResultValue = results[nestedBindingId]?.value;
-          if (nestedBindingResultValue) {
+          const nestedBindingResult = results[nestedBindingId];
+
+          if (nestedBindingResult) {
             bindingResult = updatePath(
               bindingResult,
               `value.${nestedBindingId.replace(bindingId, '')}`,
-              nestedBindingResultValue,
+              nestedBindingResult.value,
```

Comment:
Is this fixing a bug? If so, if so it could be a good on to be covered by a test (if it isn't already).